storage/reads/datatypes/storage_common.proto

```diff
@@ -179,3 +178,55 @@ message TagValuesRequest {
 message StringValuesResponse {
   repeated bytes values = 1;
 }
+
+// MeasurementNamesRequest is the request message for Storage.MeasurementNames.
+message MeasurementNamesRequest {
+  google.protobuf.Any source = 1;
+  TimestampRange range = 2 [(gogoproto.nullable) = false];
+}
+
+// MeasurementTagKeysRequest is the request message for Storage.MeasurementTagKeys.
+message MeasurementTagKeysRequest {
+  google.protobuf.Any source = 1;
+  string measurement = 2;
+  TimestampRange range = 3 [(gogoproto.nullable) = false];
+  Predicate predicate = 4;
+}
+
+// MeasurementTagValuesRequest is the request message for Storage.MeasurementTagValues.
+message MeasurementTagValuesRequest {
+  google.protobuf.Any source = 1;
+  string measurement = 2;
+  string tag_key = 3;
+  TimestampRange range = 4 [(gogoproto.nullable) = false];
+  Predicate predicate = 5;
+}
+
+// MeasurementFieldsRequest is the request message for Storage.MeasurementFields.
+message MeasurementFieldsRequest {
+  google.protobuf.Any source = 1;
+  string measurement = 2;
+  TimestampRange range = 3 [(gogoproto.nullable) = false];
+  Predicate predicate = 4;
+}
+
+message MeasurementFieldsResponse {
+  enum FieldType {
+    option (gogoproto.goproto_enum_prefix) = false;
+
+    FLOAT = 0 [(gogoproto.enumvalue_customname) = "FieldTypeFloat"];
+    INTEGER = 1 [(gogoproto.enumvalue_customname) = "FieldTypeInteger"];
+    UNSIGNED = 2 [(gogoproto.enumvalue_customname) = "FieldTypeUnsigned"];
+    STRING = 3 [(gogoproto.enumvalue_customname) = "FieldTypeString"];
+    BOOLEAN = 4 [(gogoproto.enumvalue_customname) = "FieldTypeBoolean"];
+    UNDEFINED = 5 [(gogoproto.enumvalue_customname) = "FieldTypeUndefined"];
+  }
+
+  message MessageField {
+    string key = 1;
+    FieldType type = 2;
+    sfixed64 timestamp = 3;
```

Comment:
Why this timestamp field? I expect values and tables to have timestamps.

Reply:
I called this out in the PR description, as I figured it needed clarification:

### Add maximum Timestamp to MeasurementField struct

The maximum `timestamp` captures the last timestamp observed for the specified field, which is required in order to correctly merge results from multiple nodes in a distributed environment.